nft_ingester/src/bin/synchronizer/main.rs

```diff
@@ -110,46 +99,51 @@ pub async fn main() -> Result<(), IngesterError> {
     let mut sync_tasks = JoinSet::new();
     for asset_type in ASSET_TYPES {
         let synchronizer = synchronizer.clone();
-        let shutdown_rx = shutdown_rx.resubscribe();
-        let shutdown_token = shutdown_token.clone();
-        sync_tasks.spawn(async move {
-            if let Ok(SyncStatus::FullSyncRequired(_)) = synchronizer
-                .get_sync_state(args.dump_sync_threshold, asset_type)
-                .await
-            {
-                tracing::info!("Starting full sync for {:?}", asset_type);
-                let res = synchronizer.full_syncronize(&shutdown_rx, asset_type).await;
-                match res {
-                    Ok(_) => {
-                        tracing::info!("Full {:?} synchronization finished successfully", asset_type);
-                    }
-                    Err(e) => {
-                        tracing::error!("Full {:?} synchronization failed: {:?}", asset_type, e);
+        sync_tasks.spawn({
```

Comment:
a lot of `cancellation_token` is happening here. can't it be replaced by one select! ?